### PR TITLE
Clean up css

### PR DESCRIPTION
**Pull request by jlvandenhout**
Now that we converged on a style for all elements of the Wiki, I want to take a break in restyling and first clean up the style sheets to improve ease of maintenance based on a few guide lines. These are still open for discussion, but basic ideas are:

- No styled components or CSS modules for now.
- Only use the component style attribute when dynamic styling is needed.
- Use style sheets co-located to components for basic layout and default styling.
- Use style sheets near the root of the project to override default component styling where needed (currently custom.css).
- Use CSS classes with BEM naming.
- Use the Infima style system as much as possible.

The thought behind these guidelines is to enable anyone with basic CSS knowledge to contribute, without a real need to know JS or the project structure.

Further improvements proposed:

- [x] Factor out fonts into their own style sheet.
- [x] Get consistent with BEM class naming.
- [x] Replace hardcoded values with Infima variables where possible.
- [x] Separate variable definitions from actual styling into separate files.
- [x] Adopt a CSS code format (to be discussed).
- [x] Adjust the resulting style sheets to reflect the current design agreed upon.
- [x] Fix sidebar spacing.
- [x] Fix font weights.
- [x] Fix link icon weight.

### CODE REVIEW

File: src/css/custom.css [hunk 1]
```diff
@@ -382,133 +101,111 @@ html[data-theme='dark'] {
...
+}
 
-  /*
-  Patch to avoid displaying a double header in imported DOCS
-  */
-  div.markdown > p > header > h1 {
+/*
+Patch to avoid displaying a double header in imported DOCS
+*/
+.markdown header h1 {
```
Comment (Critical94) [inline src/css/custom.css:209]: This change removes every h1 headline in the docs. That was not the idea?

Comment (jlvandenhout) [reply src/css/custom.css]: Ah excuse me, got a little too enthusiastic there I think!

File: src/css/fonts.css [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  url('fonts/icomoon.svg?yl49be#icomoon') format('svg');
+  font-weight: normal;
+  font-style: normal;
+  font-display: block;
+}
+
+/**
+Text fonts
+**/
+@import url('https://fonts.googleapis.com/css2?family=Inter:wght@300;400;500;600;700&display=swap');
```
Comment (lucas-tortora) [inline src/css/fonts.css:19]: I'm not sure why, but my IDE tells me this should be at the top of the file

File: src/theme/NavbarItem/styles.css [hunk 3]
```diff
@@ -1,63 +1,65 @@
...
+}
+
+.dropdown__container {
+  position: absolute;
+  padding-top: calc(var(--ifm-navbar-padding-vertical) + var(--ifm-global-border-width) + 5px);
   width: 100%;
   opacity: 0;
   visibility: hidden;
   z-index: var(--ifm-z-index-dropdown);
   position: absolute;
```
Comment (lucas-tortora) [inline src/theme/NavbarItem/styles.css:18]: duplicate property

File: src/theme/Social/index.js [hunk 4]
```diff
@@ -57,23 +56,23 @@ const SocialLinkList = [
 function SocialLink({ title, imageUrl, url, backgroundColor }) {
   return (
     <a
-    className={clsx('padding-horiz--sm padding-vert--md', styles.socialLink)}
+    className='social__link padding-horiz--sm padding-vert--md'
     style={{backgroundColor: backgroundColor}}
     href={url}>
-      <img className={clsx(styles.socialImage)} src={useBaseUrl(imageUrl)} />
-      <div className={clsx(styles.socialTitle)}>{title}</div>
+      <img className='social__image' src={useBaseUrl(imageUrl)} />
```
Comment (lucas-tortora) [inline src/theme/Social/index.js:62]:
```suggestion
      <img alt={title} className='social__image' src={useBaseUrl(imageUrl)} />
```

**Review by lucas-tortora — COMMENTED** [review]
Some small improvements, but it looks great :)

**Review by Critical94 — APPROVED** [review]
Added link decoration on hover (underline) as discussed a few days ago

File: src/css/custom.css [hunk 5]
```diff
@@ -1,518 +1,160 @@
...
+  color: var(--ifm-menu-color);
 }
 
+::selection {
+  background-color: #b3d4fc;
+  color: #000;
+  text-shadow: none;
+}
 
 /**
```
[the remark this answers is not in the document]
Comment (jlvandenhout) [reply src/css/custom.css:159]: We thought of `patches.css` to be a file where we patch 'logic' of the Docusaurus style system, while `custom.css` is for overriding default styling. As this essentially overrides default browser behavior I think it is okay to leave it here.

File: src/components/LandingpageHeader/index.js [hunk 6]
```diff
@@ -80,24 +53,23 @@ function Action({ title, image, image_hover, link, description }) {
 
 function LandingpageHeader() {
```
[the remark this answers is not in the document]
Comment (jlvandenhout) [reply src/components/LandingpageHeader/index.js:54]: I replaced all occurrences to `LandingPageHeader`.

File: src/css/fonts.css (hunk 2 above)
Comment (jlvandenhout) [reply src/css/fonts.css:19]: I think your IDE is right as per [the docs](https://developer.mozilla.org/en-US/docs/Web/CSS/@import). I'll correct this.

**Review by Dr-Electron — APPROVED** [review]

**Pull request merged**